pkg/platform/api/buildplanner/model/buildplan.go

```diff
@@ -148,11 +148,18 @@ func (b *BuildPlanByProject) Build() (*Build, error) {
 		var errs []string
 		var isTransient bool
 		for _, se := range b.Project.Commit.Build.SubErrors {
-			errs = append(errs, se.Message)
-			isTransient = se.IsTransient
+			if se.Message != "" {
+				errs = append(errs, se.Message)
+				isTransient = se.IsTransient
+			}
+			for _, ve := range se.ValidationErrors {
+				if ve.Error != "" {
+					errs = append(errs, ve.Error)
+				}
+			}
 		}
 		return nil, &BuildPlannerError{
-			Wrapped:          locale.NewInputError("err_buildplanner", b.Project.Commit.Build.Message),
+			Wrapped:          locale.NewInputError("err_buildplanner"),
```

Comment:
I think the idea was that the user requested something that the solver couldn't solve... This was originally a `SolverError` type that was repurposed for BuildPlanner.

Reply:
But can we definitively say that's the issue and not something non-userinput related? I don't think we can. Either we need to make an effort to identify user input or we shouldn't mark this as user input. 

End of the day we want to avoid these two scenario's:

* Rollbar gets flooded with user input errors.
* We don't get rollbar errors reported that happened due to an internal issue.

I feel like currently we are hitting the second bullet point. But just changing this to a non-user input error would probably make us hit the first. So we might need more sophistication here?

Reply:
I thought user input errors weren't reported to rollbar? Has that changed?

That said, in the current changeset I've completely removed the `Wrapped` field and the `Error()` function returns a string with all of the errors we encountered in other fields of the build plan. Do we want to introduce it back?

Reply:
I think this means that rollbar reporting would be handled further up the chain by the callers of this function.